Network/Socket/Options.hsc

```diff
@@ -376,6 +389,60 @@ getSockOpt s (SockOpt level opt) = do
                 c_getsockopt fd level opt ptr ptr_sz
         peek ptr
 
+
+socketOptionPairs :: [Pair SocketOption String]
+socketOptionPairs =
+    [ (UnsupportedSocketOption, "Unsupported")
+    , (Debug, "Debug")
+    , (ReuseAddr, "ReuseAddr")
+    , (SoDomain, "SoDomain")
+    , (Type, "Type")
+    , (SoProtocol, "SoProtocol")
+    , (SoError, "SoError")
+    , (DontRoute, "DontRoute")
+    , (Broadcast, "Broadcast")
+    , (SendBuffer, "SendBuffer")
+    , (RecvBuffer, "RecvBuffer")
+    , (KeepAlive, "KeepAlive")
+    , (OOBInline, "OOBInline")
+    , (Linger, "Linger")
+    , (ReusePort, "ReusePort")
+    , (RecvLowWater, "RecvLowWater")
+    , (SendLowWater, "SendLowWater")
+    , (RecvTimeOut, "RecvTimeOut")
+    , (SendTimeOut, "SendTimeOut")
+    , (UseLoopBack, "UseLoopBack")
+    , (MaxSegment, "MaxSegment")
+    , (NoDelay, "NoDelay")
+    , (UserTimeout, "UserTimeout")
+    , (Cork, "Cork")
+    , (TimeToLive, "TimeToLive")
+    , (RecvIPv4TTL, "RecvIPv4TTL")
+    , (RecvIPv4TOS, "RecvIPv4TOS")
+    , (RecvIPv4PktInfo, "RecvIPv4PktInfo")
+    , (IPv6Only, "IPv6Only")
+    , (RecvIPv6HopLimit, "RecvIPv6HopLimit")
+    , (RecvIPv6TClass, "RecvIPv6TClass")
+    , (RecvIPv6PktInfo, "RecvIPv6PktInfo")
+    ]
+
+socketOptionBijection :: Bijection SocketOption String
+socketOptionBijection = Bijection{..}
+    where
+        cso = "CustomSockOpt"
+        defFwd = \(CustomSockOpt (n,m)) -> cso++show n++"_"++show m
```

Comment:
I would like have a space (`'  '`) before/after `++`.

Reply:
Leaving out spaces ensures that the `read` conversion only needs to see a single token. While the style would be improved, it would be difficult to implement without requiring more invasive changes on the other side.

Reply:
Unless you meant space characters in the source rather than space characters in the function result.

Reply:
I meant `cso ++ show n ++ "_" ++ show m`. But this is not mandatory. :-)